lib/cache/cache.go

```diff
@@ -593,6 +616,9 @@ const (
 
 // New creates a new instance of Cache
 func New(config Config) (*Cache, error) {
+	if err := utils.RegisterPrometheusCollectors(cacheCollectors...); err != nil {
```

Comment:
Hmm, I realize this is a backport, but registering collectors is a one-time thing right?

I don't like the idea of mutating global state in a constructor like this. What happens if I call `New` twice? Does the second attempt fail?

Reply:
With regards to registering collectors only once, I would say "usually". You are allowed to register the same metric twice but will return an error if fully qualified name of the metric (namespace, subsystem, name) and labels are inconsistent on the subsequent registrations. This call is fine because the name and labels do not change between registration calls [[docs](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#hdr-Advanced_Uses_of_the_Registry)]. 

Here is a snippet from Teleport instance running auth, proxy, kube, and apps service at the same time:
```
# HELP teleport_cache_events Number of events received by a Teleport service cache. Teleport's Auth Service, Proxy Service, and other services cache incoming events related to the
ir service.
# TYPE teleport_cache_events counter
teleport_cache_events{cache_component="apps"} 8
teleport_cache_events{cache_component="auth"} 43
teleport_cache_events{cache_component="kube"} 8
teleport_cache_events{cache_component="proxy"} 49
teleport_cache_events{cache_component="remote-proxy"} 50
```
Each service calls `New()` with proxy calling it twice.

Reply:
This actually a lazier loading than using the `MustRegister` in an `init()` function as that would be called every time the package is imported. IIRC the `init()` pattern was used earlier but changed to struct creation since a lot of unused metrics were being registered unnecessarily (particularly db related metrics).